controller/src/main/java/org/jboss/as/controller/operations/global/ReadResourceHandler.java

```diff
@@ -326,8 +327,16 @@ private void doExecuteInternal(OperationContext context, ModelNode operation, bo
                                 childMap.setEmptyObject();
                                 directChildren.put(childType, childMap);
                             }
-                            // Add a "child" => undefined
-                            childMap.get(child);
+                            // In case of runtime resources adds '=> undefined' if there's no include-runtime parameter,
+                            // in read-resource operation, otherwise adds '{"child" => undefined}'
+                            if (childReg.isRuntimeOnly()){
+                                if (queryRuntime){
+                                    childMap.get(child);
+                                }
+                            }
+                            else {
+                                childMap.get(child);
+                            }
```

Comment:
This can be reduced to

```
if (queryRuntime || !childReg.isRuntimeOnly()) { 
  childMap.get(child);
}
```

Reply:
Actually, it needs to be null safe, so...

```
if (queryRuntime || (childReg != null && !childReg.isRuntimeOnly())) { 
  childMap.get(child);
}
```

Reply:
Thank you, code is looking better now